### PR TITLE
Fix: Action links incorrectly enabled on empty table (DFCT0010080)

### PR DESCRIPTION
**Pull request by ericenns**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR fixes the Action Links on Project and Group samples from being incorrectly enabled when No Samples associated with a Group or a Project.

This PR also refactors the Group and Project Samples tables.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

image

image

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Create an new Project without any Samples
2. Navigate to the Samples page for the new Project
3. Ensure that the action links are disabled and that an empty state message is shown
4. Create a new Group
5. Add the Project created above to the group
6. Navigate to the Samples page for the new Group
7. Ensure that the action links are disabled and that an empty state message is shown

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

**Review by joshsadam — CHANGES_REQUESTED** [review]
This looks awesome, just one request

File: app/components/samples/table_component.html.erb [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+            <td
+              class="
+                px-6 py-3 sticky left-0 bg-slate-50 dark:bg-slate-900 z-10
+              "
+              colspan="3"
+            >
+              <span>
+                <%= t(".counts.samples") %>:
+                <strong data-selection-target="total">0</strong>
+              </span>
```
[the remark this answers is not in the document]
Comment (ericenns) [reply app/components/samples/table_component.html.erb:214]: Fixed in [0cbf19d](https://github.com/phac-nml/irida-next/pull/635/commits/0cbf19d4d18c500cbcca7648bcbe238216f434d8)[0cbf19d](https://github.com/phac-nml/irida-next/pull/635/commits/0cbf19d4d18c500cbcca7648bcbe238216f434d8)

[the remark this answers is not in the document]
Comment (joshsadam) [reply app/components/samples/table_component.html.erb:214]: This looks good, things

File: app/components/samples/table_component.fr.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+en:
```
Comment (ChrisHuynh333) [inline app/components/samples/table_component.fr.yml:1]: I believe this should be `fr:`

**Review by ChrisHuynh333 — CHANGES_REQUESTED** [review]
Looks good so far, just one from me currently

**Review by joshsadam — DISMISSED** [review]
Looks great, thanks :)

File: app/components/samples/table_component.fr.yml (hunk 2 above)
Comment (ericenns) [reply app/components/samples/table_component.fr.yml]: Fixed in 37689b5

**Review by ChrisHuynh333 — APPROVED** [review]
Looks great!

**Review by deepsidhu85 — DISMISSED** [review]
Looks good to me!

**Review by deepsidhu85 — APPROVED** [review]

**Pull request merged**